working/declaring-constructors/feature-specification.md

```diff
@@ -329,12 +439,79 @@ It is a compile-time error if:
 
 *   A field parameter is named `_`. *We could allow this but... why?*
 
+### Private field parameters and initializing formals
+
+An identifier is a *private name* if it starts with an underscore (`_`),
+otherwise it's a *public name*.
+
+A private name may have a *corresponding public name*. If the characters of the
+identifier with the leading underscore removed form a valid identifier and a
+public name, then that is the private name's corresponding public name. *For
+example, the corresponding public name of `_foo` is `foo`.* If removing the
+underscore does not leave something which is is a valid identifier *(as in `_`
+or `_2x`)* or leaves another private name *(as in `__x`)*, then the private name
+has no corresponding public name.
+
+The private declared name, *p*, of an initializing formal or field parameter in
+constructor C has a corresponding *non-conflicting public name* if it has a
+corresponding public name, *n*, and no other parameter of the same constructor
+declaration has either of the names *p* or *n* as declared name. *In other
+words, if removing the `_` leads to a collision with another parameter, then
+there is a conflict.*
```

Comment:
(And also if not removing it leads to a conflict, `foo(this._foo, int _foo);` is a conflict because the local `_foo` names clash in the initializer list scope. It's always a clash if two parameters have the same name. Now it's also a "clash" if the corresponding public name of a privately named parameter is the same as the name of another parameter. It won't necessarily cause an error, it may just make the corresponding public name not be used: `Foo(this._foo, int foo):` is valid today, and will stay valid with a privately named first parameter.)